protocols/gossipsub/src/handler.rs

```diff
@@ -274,7 +274,13 @@ impl ProtocolsHandler for GossipsubHandler {
                                         Some(OutboundSubstreamState::PendingFlush(substream))
                                 }
                                 Err(e) => {
-                                    return Poll::Ready(ProtocolsHandlerEvent::Close(e));
+                                    if let io::ErrorKind::PermissionDenied = e.kind() {
+                                        error!("Message over the maximum transmission limit was not sent.");
```

Comment:
Am I right assuming that you encountered this using secio [here](https://github.com/libp2p/rust-libp2p/blob/master/protocols/secio/src/codec/len_prefix.rs#L60)?

I am not sure it is safe to assume that `PermissionDenied` is always related to maximum transmission limit failures. Thus I would suggest including the original error message here.

I do think that `PermissionDenied` always implies a mistake on the local node's end, thus not disconnecting sounds good to me.

Reply:
I can't recall, I use secio and noise often. 

The `PermissionDenied` is an error that the UVICodec emits when we go over the maximum size. This is independent of the substream type, transport or authentication layer. 

What is happening, is the message gets sent to the `Framed` stream. It uses the GossipsubCodec to encode the data. The UVICodec realises the encoded message is too large and returns `PermissionDenied`. This will always occur regardless of any underlying stream.

This also happens locally, no information is sent to the underlying substream.